analyses/mutational-signatures/04-analyze_de_novo.Rmd

```diff
@@ -0,0 +1,175 @@
+  # This takes a few minutes to parse.
+  de_novo_signatures <- list()
+  de_novo_exposures <- list()
+  for (this_model in c("poisson" , "multinomial"))
+  {
+    for (this_seed in 1:5){
+  
+      this_name <- glue::glue("seed_{this_seed}_model_{this_model}")
+      filename <- file.path(path_to_input_rds, 
+                            glue::glue("{this_name}.RDS"))
+      #cat(filename)
+      if (file.exists(filename)) {
+        fit <- readr::read_rds(filename)
+        for (k in 2:5){
+          sig_name <- glue::glue("{k}_{this_name}")
+          de_novo_signatures[[ sig_name ]] <- sigfit::retrieve_pars(fit[[k]], par = "signatures") 
+          de_novo_exposures[[ sig_name ]] <- sigfit::retrieve_pars(fit[[k]], par = "exposures") 
+        }
+      }
+    }
+    
+  }
+  # We save this list to a result file
+  readr::write_rds(de_novo_signatures, de_novo_signatures_file)
+  readr::write_rds(de_novo_exposures, de_novo_exposures_file)
+}
+```
+
+<br><br>
+Based on goodness of fit analyses, we expect these *k* to be optimal for the given seed and inference model (noting that we do see strong sensitivity to this starting condition).
+
+```{r gof_k}
+gof <- tibble::tribble(
+  ~model,     ~seed,      ~best_k,
+  #-----------------
+  "poisson",1,3,
+  "multinomial",1,3,
+  "poisson",2,4,
+  "multinomial",2,4,
+  "poisson",3,3,
+  "multinomial",3,3,
+  "poisson",4,3,
+  "multinomial",4,3,  
+  "poisson",5,3,
+  "multinomial",5,3  
+)
+```
+
+
+<br><br>
+Now, we can determine which signatures were extracted and map back to COSMIC names using cosine similarity to determine matching signatures:
+
+```{r determine_sigs}
+
+# Will find the COSMIC equivalent using cosine similarity
+map_match <- function(x) {
+  sort( # Can sort since we just want to see cosmic matches
+    as.numeric( # none of this "optimal assigment ==> 1" business
+      match_signatures(x, cosmic_signatures_v3)
+    )
+  )
+}
+
+purrr::map(de_novo_signatures, map_match) -> cosmic_matches
+```
+
+<br><br>
+Now, we create a table of results for which COSMIC SBS signatures were extracted. **Full results for each combination of k, model, and random seed:**
+```{r sigs_full}
+tibble::tibble(
+  name = names(cosmic_matches), 
+  cosmic_sig_index = unname(cosmic_matches[1:length(cosmic_matches)])
+  ) %>%
+  tidyr::unnest(cosmic_sig_index) %>%
+  dplyr::mutate(cosmic_sig = cosmic_names[cosmic_sig_index]) -> unnested_sigs
+
+unnested_sigs %>%
+  dplyr::group_by(name) %>%
+  dplyr::summarize(cosmic_sigs = toString(cosmic_sig)) %>%
+  dplyr::mutate(cosmic_sigs = stringr::str_replace_all(cosmic_sigs, "SBS", "")) %>%
+  dplyr::mutate(seed  = stringr::str_extract(name, "_\\d_"),
+                seed  = stringr::str_replace_all(seed, "_", ""),
+                k     = stringr::str_extract(name, "^\\d"), 
+                model = stringr::str_extract(name, "[a-z]+$")) %>%
+  dplyr::select(-name) -> full_sigs
+
+full_sigs %>%
+  tidyr::spread(seed, cosmic_sigs) %>%
+  knitr::kable() 
```

Comment:
This (and the one below) isn't showing up in the rendered HTML version when I check out this branch locally.

Reply:
I swear, I am incapable of this. `.html` is fine, but with...`nb.html`, run chunks as I might, output eludes me.

Reply:
Doesn't need to be a notebook necessarily!